tests/models/kosmos2/test_modeling_kosmos2.py

```diff
@@ -762,3 +762,27 @@ def test_snowman_image_captioning_batch(self):
         self.assertEqual(processed_text[0], EXPECTED_PROCESSED_TEXT_0)
         self.assertEqual(all_final_text[0], EXPECTED_FINAL_TEXT_0)
         self.assertListEqual(all_entities[0], EXPECTED_ENTITIES_0)
+
+    @slow
+    def test_inference_interpolate_pos_encoding(self):
+        # ViT models have an `interpolate_pos_encoding` argument in their forward method,
+        # allowing to interpolate the pre-trained position embeddings in order to use
+        # the model on higher resolutions. The DINO model by Facebook AI leverages this
+        # to visualize self-attention on higher resolution images.
+        model = Kosmos2Model.from_pretrained("microsoft/kosmos-2-patch14-224").to(torch_device)
+
+        processor = AutoProcessor.from_pretrained(
+            "microsoft/kosmos-2-patch14-224", padding_side="left", size={"shortest_edge": 480}
+        )
+
+        image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+        inputs = processor(text="what's in the image", images=image, return_tensors="pt").to(torch_device)
+
+        # forward pass
+        with torch.no_grad():
+            outputs = model(**inputs, interpolate_pos_encoding=True)
```

Comment:
Same here re failing

Reply:
Hy @amyeroberts , I am getting the same results in `model` with  `interpolate_pos_encoding=True` or `False` in both x_clip or kosmos2. No error, just the same exact tensor for `outputs.vision_model_output.last_hidden_state`.

Reply:
I think the reason is because a logic like this:

``` if not interpolate_pos_encoding and (height != self.image_size[0] or width != self.image_size[1]):
            raise ValueError(
                f"Input image size ({height}*{width}) doesn't match model"
                f" ({self.image_size[0]}*{self.image_size[1]})."
            )
```
is not implemented in the `modeling_kosmos2.py` file and its corresponding one in `modeling_x_clip.py`. 

Happy to implement it.

Reply:
Part of the issue is originated, in the case of the model kosmos2, because 

```
        processor = AutoProcessor.from_pretrained(
            "microsoft/kosmos-2-patch14-224", padding_side="left", size={"shortest_edge": 480}
        )
```
is not changing the image size. It is still returning a 224x224 image, even when you reduce the shortest_edge to less than 224.

Need to investigate further.

Reply:
I've found out what was missing:
```
        # default imnage size of pretrained kosmos_2 is 224 x 224
        processor = AutoProcessor.from_pretrained(
            "microsoft/kosmos-2-patch14-224", size={"shortest_edge": 180}, crop_size = {'height': 180, 'width': 180}
        )
```

Now, I have implemented the ValueError in the `modeling_kosmos2.py` file.